docs/docs/modules/builders/generating_transactions.mdx

```diff
@@ -0,0 +1,121 @@
+Any change made to the builder can impact the size and thus the fee so the fee should be the last thing set.
+If implicitly setting the fee any extra ADA (`inputs + withdrawals - outputs + refund - deposit - min fee`) is sent to the provided change address.
+Fees must be sufficient, i.e. `inputs + withdrawals + refund >= outputs + deposit + fee` which must be manually ensured if you explicitly set the fee. Any extra fee is not necessary and the extra ADA beyond that will be burned.
+Once the transaction is ready, `const body = builder.build()` can be called to return a ready `TransactionBody`.
+
+Withdrawals are ADA withdrawn as part of the rewards generated by staking and deposits are refundable ADA locked while resources such as stake certificates or pool registrations exist on the blockchain. They are returned as refunds when these resources are deregistered/retired.
+
+To get to a transaction ready to post on the blockchain, we must create a `Transaction` from that, which consists of the `TransactionBody`, a matching `TransactionWitnessSet` and optionally a `TransactionMetadata`.
+The witnesses and optional metadata must match those provided to the builder. The witnesses must sign the hash of the transaction body returned by `hash_transaction(body)`. In addition to the witnesses for inputs, withdrawals and some certificates require witnesses as well. For example, staking address registration does not require a witness while stake address de-registration requires one. For any questions or doubts about the rules governing fees, deposits, rewards, certificates or which witness types are required refer to the [shelley specs](https://github.com/input-output-hk/cardano-ledger-specs#cardano-ledger), specifically the Shelley design specification for general design. The formal specification could be useful for specific details as well. The design spec contains details about which certificates require which type of witnesses in the Certificates and Registrations section.
+
+## Example code
+
+The example below builds a transaction with all 2 of the 3 input types: key and bootstrap.
+Multisig (script) inputs are essentially identical to key inputs, but using the scripthash instead of the keyhash, however they are not supported for implicit fee calculation yet.
+Fees are automatically calculated and sent to a change address in the example.
+
+
+```javascript
+// instantiate the tx builder with the Cardano protocol parameters - these may change later on
+const linearFee = CardanoWasm.LinearFee.new(
+    CardanoWasm.BigNum.from_str('44'),
+    CardanoWasm.BigNum.from_str('155381')
+);
+const txBuilderCfg = CardanoWasm.TransactionBuilderConfigBuilder.new()
+    .fee_algo(linearFee)
+    .pool_deposit(CardanoWasm.BigNum.from_str('500000000'))
+    .key_deposit(CardanoWasm.BigNum.from_str('2000000'))
+    .max_value_size(4000)
+    .max_tx_size(8000)
+    .coins_per_utxo_word(CardanoWasm.BigNum.from_str('34482'))
```

Comment:
the API has been changed. this code must've been from a previous era as we don't have coins_per_utxo_word anymore. There's a coins_per_utxo_byte but I'm guessing there are other areas of this code that needs updating too.